src/zenml/orchestrators/step_runner.py

```diff
@@ -177,12 +187,19 @@ def run(
                     output_name
                 ].materializer_source
 
-                self._store_output_artifact(
-                    materializer_class=materializer_class,
-                    materializer_source=materializer_source,
-                    artifact=output_artifacts[output_name],
-                    data=return_value,
+                uri = output_artifact_uris[output_name]
```

Comment:
Again, I would suggest we try to create additional small methods instead of adding all code into (already far too huge) methods.

Reply:
Not as easy here as for the other one. I tried to quickly refactor some code into a util method similar as we had before but ended up refactoring the entire `StepRunner` 😄